cmd/apiserver/main.go

```diff
@@ -12,91 +12,43 @@ import (
-	SERVER_PORT_DEFAULT                = 9898
-
-	ENV_LOAD_INTERVAL_SECONDS      = "LOAD_INTERVAL_SECONDS"
-	ENV_DAILY_METRICS_TTL_SECONDS  = "DAILY_METRICS_TTL_SECONDS"
-	ENV_TODAYS_METRICS_TTL_SECONDS = "TODAYS_METRICS_TTL_SECONDS"
-	ENV_MAX_ARCHIVE_AGE_DAYS       = "MAX_ARCHIVE_AGE"
-	ENV_SERVER_PORT                = "API_SERVER_PORT"
-	ENV_BACKEND_API_URL            = "BACKEND_API_URL"
-	ENV_BACKEND_API_TOKEN          = "BACKEND_API_TOKEN"
-)
-
-type options struct {
-	loadInterval            int
-	dailyMetricsTTLSeconds  int
-	todaysMetricsTTLSeconds int
-	maxPastDays             int
-	port                    int
-	backendApiUrl           string
-	backendApiToken         string
-}
-
-func gatherOptions() (options, error) {
-	options := options{}
-
-	optsItems := []struct {
-		value        *int
-		defaultValue int
-		envVar       string
-	}{
-		{value: &options.loadInterval, defaultValue: LOAD_INTERVAL_DEFAULT_SECONDS, envVar: ENV_LOAD_INTERVAL_SECONDS},
-		{value: &options.dailyMetricsTTLSeconds, defaultValue: DAILY_METRICS_TTL_DEFAULT_SECONDS, envVar: ENV_DAILY_METRICS_TTL_SECONDS},
-		{value: &options.todaysMetricsTTLSeconds, defaultValue: TODAYS_METRICS_TTL_DEFAULT_SECONDS, envVar: ENV_TODAYS_METRICS_TTL_SECONDS},
-		{value: &options.maxPastDays, defaultValue: MAX_ARCHIVE_AGE_DEFAULT_DAYS, envVar: ENV_MAX_ARCHIVE_AGE_DAYS},
-		{value: &options.port, defaultValue: SERVER_PORT_DEFAULT, envVar: ENV_SERVER_PORT},
-	}
-
-	for _, o := range optsItems {
-		value, err := cmd.GetIntFromEnv(o.envVar, o.defaultValue)
-		if err != nil {
-			return options, err
-		}
-		*o.value = value
-	}
-
-	backendUrl := os.Getenv(ENV_BACKEND_API_URL)
-	if backendUrl == "" {
-		return options, fmt.Errorf("Missing required environment variable: %s", ENV_BACKEND_API_URL)
-	}
-	options.backendApiUrl = backendUrl
+var (
+	postgresUser     = environment.MustGetString("POSTGRES_USER")
+	postgresPassword = environment.MustGetString("POSTGRES_PASSWORD")
+	postgresDB       = environment.MustGetString("POSTGRES_DB")
+	postgresHost     = environment.MustGetString("POSTGRES_HOST")
 
-	token := os.Getenv(ENV_BACKEND_API_TOKEN)
-	if token == "" {
-		return options, fmt.Errorf("Missing required environment variable: %s", ENV_BACKEND_API_TOKEN)
-	}
-	options.backendApiToken = token
+	backendAPIURL   = environment.MustGetString("BACKEND_API_URL")
+	backendAPIToken = environment.MustGetString("BACKEND_API_TOKEN")
 
-	return options, nil
-}
+	loadInterval            = int(environment.GetInt64("LOAD_INTERVAL_SECONDS", 30))
+	dailyMetricsTTLSeconds  = int(environment.GetInt64("DAILY_METRICS_TTL_SECONDS", 120))
+	todaysMetricsTTLSeconds = int(environment.GetInt64("TODAYS_METRICS_TTL_SECONDS", 60))
+	maxPastDays             = int(environment.GetInt64("MAX_ARCHIVE_AGE", 30))
+	port                    = int(environment.GetInt64("API_SERVER_PORT", 9898))
+)
 
 type backendProvider struct {
 	db.PostgresClient
-	backendApiUrl   string
-	backendApiToken string
+	backendAPIURL   string
```

Comment:
This makes sense to do IMO. But it does not belong in this PR.
A separate PR that handles styling/naming would be much more readable.